elephant/spectral.py

```diff
@@ -523,6 +529,543 @@ def multitaper_psd(signal, n_segments=1, len_segment=None,
+    Returns
+    -------
+    freqs : np.ndarray
+        Frequencies associated with power estimate in `psd`
+    cross_spec : np.ndarray
+        Estimate of the cross spectrum of time series in `signal`
+
+    """
+    # When the input is AnalogSignal, fetch the underlying numpy array and swap
+    # axes from (n_samples, n_channels) to (n_channels, n_samples)
+    data = np.asarray(signals)
+    if isinstance(signals, neo.AnalogSignal):
+        data = np.moveaxis(data, 0, 1)
+        fs = signals.sampling_rate
+
+    # Initialize argument dictionary for multitaper_cross_spectrum called on
+    # segments
+    cross_spec_params_dict = {
+        'fs': fs,
+        'nw': nw,
+        'num_tapers': num_tapers,
+        'peak_resolution': peak_resolution,
+        'return_onesided': return_onesided,
+        'attach_units': False}
+
+    # Apply segmentation
+    freqs, cross_spec_estimate = _segmented_apply_func(
+        data=data, n_segments=n_segments, len_segment=len_segment,
+        overlap=overlap, frequency_resolution=frequency_resolution,
+        func=multitaper_cross_spectrum,
+        func_params_dict=cross_spec_params_dict)
+
+    # Attach proper units to return values
+    if isinstance(signals, pq.quantity.Quantity):
+        cross_spec_estimate = cross_spec_estimate * signals.units * \
+                              signals.units / pq.Hz
+        freqs = freqs * pq.Hz
+
+    return freqs, cross_spec_estimate
+
+
+def multitaper_coherence(signal_i, signal_j, n_segments=8, len_segment=None,
+                         frequency_resolution=None, overlap=0.5, fs=1,
+                         nw=4, num_tapers=None, peak_resolution=None):
+    """
+    Estimates the magnitude-squared coherence and phase-lag of two given
+    'neo.AnalogSignal' using the Multitaper method.
+    The function calls `segmented_multitaper_cross_spectrum` internally and
+    thus both functions share parts of the respective signatures.
+
+    .. math::
+        C(\omega)=\frac{|S_{xy}(\omega)|^2}{S_{xx}(\omega)S_{yy}(\omega)}
+
+    Parameters
+    ----------
+    signal_i : neo.AnalogSignal or pq.Quantity or np.ndarray
+        First time series data of the pair between which coherence is
+        computed.
+    signal_j : neo.AnalogSignal or pq.Quantity or np.ndarray
+        Second time series data of the pair between which coherence is
+        computed.
+        The shapes and the sampling frequencies of `signal_i` and `signal_j`
+        must be identical. When `signal_i` and `signal_j` are not
+        `neo.AnalogSignal`, sampling frequency should be specified through the
+        keyword argument `fs`. Otherwise, the default value is used
+        (`fs` = 1.0).
+    n_segments : int, optional
+        Number of segments. The length of segments is adjusted so that
+        overlapping segments cover the entire stretch of the given data. This
+        parameter is ignored if `len_segment` or `frequency_resolution` is
+        given.
+        Default: 8.
```

Comment:
Here, the default number should be 1, as the "standard" coherence (or cross-spectrum) estimation with multi-tapers is without segmentation.
In relation to this, the order of the parameters regarding segmentation should be later than those regarding multitaper itself (such as nw, num_tapers, and so on, as they specify the behavior of the core functionality of the function).

Reply:
Addressed in https://github.com/NeuralEnsemble/elephant/pull/525/commits/5ddd042c88783fdc4203819a36146c746510599d